src/util.test.ts

```diff
@@ -138,14 +138,12 @@ describe('util', () => {
       expect(response).toEqual('response');
     });
 
-    it('should timeout', () => {
-      try {
+    it('should timeout', async () => {
+      await expect(
         util.safelyExecuteWithTimeout(() => {
           return new Promise((res) => setTimeout(res, 800));
```

Comment:
Not a blocker for this PR because it was already broken in this manner, but this isn't quite enough to test that this is working correctly.

For us to know the timeout was triggered, we'd have to ensure that something other than undefined would be called otherwise. e.g.

```suggestion
          return new Promise((resolve) => setTimeout(() => resolve('No timeout!'), 800));
```

Reply:
Noted! I will fix this test in a different PR :)